src/rgw/rgw_log.cc

```diff
@@ -352,7 +352,23 @@ int rgw_log_op(RGWRados *store, RGWREST* const rest, struct req_state *s,
     set_param_str(s, "REMOTE_ADDR", entry.remote_addr);
   set_param_str(s, "HTTP_USER_AGENT", entry.user_agent);
   set_param_str(s, "HTTP_REFERRER", entry.referrer);
-  set_param_str(s, "REQUEST_URI", entry.uri);
+
+  std::string uri(s->info.env->get("REQUEST_METHOD"));
+  uri.append(" ");
+
+  uri.append(s->info.env->get("REQUEST_URI"));
+  if(s->info.env->get("QUERY_STRING") &&
+     (strlen(s->info.env->get("QUERY_STRING")) != 0)) {
```

Comment:
@mikulely @joscollin instead of calling strlen() here, should do: `*(const char *)s->info.env->get("QUERY_STRING")) != '\0'`

Reply:
@mikulely @joscollin but should only call `s->info.env->get("QUERY_STRING")` once, so put it in a temp variable first and then do the two checks and append on it